### PR TITLE
docs: correctly type getters

### PR DESCRIPTION
**Pull request by IRONM00N**
**Please describe the changes this PR makes and why it should be merged:**
- made all getters be typed correctly as getters - this improves typescript error messages and makes it clearer to the end-user that the value is calculated 
- remove `GuildChannel#calculatedPosition` from the typings since it doesn't exist in the source code
- corrected a few properties that should(n't) have been marked as readonly 

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by kyranet — APPROVED** [review]
This was a task that I wanted to get done for a while, thanks! 🙏🏼

**Review by vladfrangu — APPROVED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -2454,7 +2452,7 @@ export class Webhook extends WebhookMixin() {
   public avatar: string;
   public avatarURL(options?: ImageURLOptions): string | null;
   public channelId: Snowflake;
-  public client: Client;
+  public readonly client: Client;
```
Comment (ImRodry) [inline packages/discord.js/typings/index.d.ts:2455]: This isn't a getter but a property defined through Object.defineProperty, should it be documented this way or with the Readonly type?

Comment (IRONM00N) [reply packages/discord.js/typings/index.d.ts:2455]: Well, it isn't `writable` but it also isn't a getter so it should be like this. I think Readonly would only be used with `Object.freeze`

Comment (kyranet) [reply packages/discord.js/typings/index.d.ts:2455]: That's correct, `Object.defineProperty` by default assigns readonly properties.

**Review by iCrawl — APPROVED** [review]

**Pull request merged**